src/coreclr/vm/dllimport.cpp

```diff
@@ -2738,16 +2747,6 @@ void PInvokeStaticSigInfo::DllImportInit(_In_ MethodDesc* pMD, _Outptr_opt_ LPCU
     mdModuleRef modref = mdModuleRefNil;
     if (FAILED(pInternalImport->GetPinvokeMap(pMD->GetMemberDef(), (DWORD*)&mappingFlags, ppEntryPointName, &modref)))
     {
-#if !defined(CROSSGEN_COMPILE) // IJW
-        // The guessing heuristic has been broken with NGen for a long time since we stopped loading
-        // images at NGen time using full LoadLibrary. The DLL references are not resolved correctly
-        // without full LoadLibrary.
-        //
-        // Disable the heuristic consistently during NGen so that it does not kick in by accident.
-        if (!IsCompilationProcess())
-            BestGuessNDirectDefaults(pMD);
```

Comment:
I'm missing where this was moved.

Reply:
It was just deleted. `BestGuessNDirectDefaults` was simply finding the would-be target of the call and then doing nothing with it.

It used to check if the target looked like a Win32 API call and set `nlfLastError` if it did. That stopped being necessary long ago - the VC linker handles it (in the Whidbey timeframe based on comments in history). The portion setting that flag was removed, but the lead-up to it wasn't.